sklearn/feature_selection/tests/test_rfe.py

```diff
@@ -649,3 +649,47 @@ def test_rfe_estimator_attribute_error():
         rfe.fit(iris.data, iris.target).decision_function(iris.data)
     assert isinstance(exec_info.value.__cause__, AttributeError)
     assert inner_msg in str(exec_info.value.__cause__)
+
+
+@pytest.mark.parametrize(
+    "ClsRFE, param, warning_msg",
+    [
+        (
+            RFE,
+            "n_features_to_select",
+            (
+                "Found n_features_to_select=21 > n_features=20."
+                " There will be no feature selection and all features will be kept."
+            ),
+        ),
+        (
+            RFECV,
+            "min_features_to_select",
+            (
+                "Found self.min_features_to_select=21 > n_features=20."
+                " There will be no feature selection and all features will be kept."
+            ),
+        ),
+    ],
+)
+def test_rfe_n_features_to_select_warning(ClsRFE, param, warning_msg):
+    """Check if the correct warning is raised when trying to initialize a RFE
+    object with a n_features_to_select attribute larger than the number of
+    features present in the X variable that is passed to the fit method
+
+    Non-regression test for:
+    https://github.com/scikit-learn/scikit-learn/issues/28725
+    """
+    X, y = make_classification(n_features=20, random_state=0)
+
+    with pytest.warns(UserWarning, match=f"{param}=21 > n_features=20") as record:
```

Comment:
Checking that part of the message is enough, no need to add the full message in the parametrization.
```suggestion
    with pytest.warns(UserWarning, match=f"{param}=21 > n_features=20"):
```